x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/chart_switch.tsx

```diff
@@ -222,89 +310,73 @@ export const ChartSwitch = memo(function ChartSwitch(props: Props) {
-            betaBadgeLabel={
-              v.selection.dataLoss !== 'nothing'
-                ? i18n.translate('xpack.lens.chartSwitch.dataLossLabel', {
-                    defaultMessage: 'Data loss',
-                  })
-                : undefined
-            }
-            betaBadgeTooltipContent={
-              v.selection.dataLoss !== 'nothing'
-                ? i18n.translate('xpack.lens.chartSwitch.dataLossDescription', {
-                    defaultMessage: 'Switching to this chart will lose some of the configuration',
-                  })
-                : undefined
-            }
-            betaBadgeIconType={v.selection.dataLoss !== 'nothing' ? 'alert' : undefined}
+  return (
+    <div className="lnsChartSwitch__header">
+      <EuiPopover
+        id="lnsChartSwitchPopover"
+        ownFocus
+        initialFocus=".lnsChartSwitch__popoverPanel"
+        panelClassName="lnsChartSwitch__popoverPanel"
+        panelPaddingSize="s"
+        button={
+          <ToolbarButton
+            onClick={() => setFlyoutOpen(!flyoutOpen)}
+            data-test-subj="lnsChartSwitchPopover"
+            fontWeight="bold"
           >
-            <EuiIcon className="lnsChartSwitch__chartIcon" type={v.icon || 'empty'} size="l" />
-          </EuiKeyPadMenuItem>
-        ))}
-      </EuiKeyPadMenu>
-      {searchTerm && (visualizationTypes || []).length === 0 && (
-        <EuiSelectableMessage>
-          <FormattedMessage
-            id="xpack.lens.chartSwitch.noResults"
-            defaultMessage="No results found for {term}."
-            values={{
-              term: <strong>{searchTerm}</strong>,
-            }}
-          />
-        </EuiSelectableMessage>
-      )}
-    </EuiPopover>
+            <VisualizationSummary {...props} />
+          </ToolbarButton>
+        }
+        isOpen={flyoutOpen}
+        closePopover={() => setFlyoutOpen(false)}
+        anchorPosition="downLeft"
+      >
+        <EuiPopoverTitle>
+          <EuiFlexGroup alignItems="center" responsive={false}>
+            <EuiFlexItem>
+              {i18n.translate('xpack.lens.configPanel.chartType', {
+                defaultMessage: 'Chart type',
+              })}
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiPopoverTitle>
+        <EuiSelectable
+          height={computeListHeight(visualizationTypes, MAX_LIST_HEIGHT)}
+          searchable
+          singleSelection
+          isPreFiltered
+          data-test-subj="lnsChartSwitchList"
+          searchProps={{
+            incremental: true,
+            className: 'lnsChartSwitch__search',
+            'data-test-subj': 'lnsChartSwitchSearch',
+            onSearch: (value) => setSearchTerm(value),
+          }}
+          options={visualizationTypes}
+          onChange={(newOptions) => {
+            const id = newOptions.find(({ checked }) => checked === 'on')!.value!;
+            commitSelection(visualizationsLookup[id].selection);
+          }}
```

Comment:
We've had this bug before, but this appears to be a pretty big one. Lens crashes if you select the currently selected option.